http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpClient.java

```diff
@@ -32,4 +32,5 @@
      */
     AbortableCallable<SdkHttpFullResponse> prepareRequest(SdkHttpFullRequest request, SdkRequestContext requestContext);
 
+    String clientName();
```

Comment:
I can add documentation but their will always be the possibility of skewed metrics. I think having it potentially inaccurate is still better than not having the data. Hopefully people will use good values if well documented.

Will add sanitization.